hs_core/tests/api/native/test_publish_resource.py

```diff
@@ -35,20 +32,40 @@ def setUp(self):
         )
 
     def tearDown(self):
+        super(TestPublishResource, self).tearDown()
         self.user.uaccess.delete()
         User.objects.all().delete()
         Group.objects.all().delete()
         GenericResource.objects.all().delete()
 
     def test_publish_resource(self):
-        # publish resource
+        # check status prior to publishing the resource
+        self.assertFalse(
+            self.res.raccess.published,
+            msg='The resource is published'
+        )
+
+        self.assertFalse(
+            self.res.raccess.immutable,
+            msg='The resource is frozen'
+        )
+
+        self.assertIsNone(
+            self.res.doi,
+            msg='doi is assigned'
+        )
+
+        # there should not be published date type metadata element
+        self.assertFalse(self.res.metadata.dates.filter(type='published').exists())
+
+        # publish resource - this is the api we are testing
         hydroshare.publish_resource(self.res.short_id)
```

Comment:
@pkdash Once resource publication is fully implemented and in production, this publish_resource() function will register the resource doi via crossref which will cost us money once in production, so we cannot call this function to test resource publication. So I am wondering whether we really need to test resource publication given its unique money-causing characteristic. I could add something into all publication related functions to use crossref test system rather than crossref real doi site, but it'll take some extra effort to do so especially considering we will have a background celery task running to check each resource submitted for publication. I am copying @dtarb @horsburgh @ironfroggy for their input. I am leaning toward skipping resource publication test for now, but could spend some extra time to put crossref test system in as well for testing purposes if needed.